src/shell/Shell.cpp

```diff
@@ -652,12 +652,16 @@ int main(int argc, char* argv[])
                 }
                 if (strcmp(argv[i], "--debugger-wait-source") == 0) {
                     StringRef* sourceName;
-                    StringRef* clientSourceRef = context->getClientSource(&sourceName);
-                    if (clientSourceRef) {
-                        if (!evalScript(context, clientSourceRef, sourceName, false, false))
-                            return 3;
-                        runShell = false;
-                        break;
+                    StringRef* clientSourceRef;
+                    while (true) {
+                        clientSourceRef = context->getClientSource(&sourceName);
+                        if (clientSourceRef) {
```

Comment:
I would use 
```
if (clientSourceRef) {
  break;
}

if (!evalScript(context, clientSourceRef, sourceName, false, false))
...
```

This requires less indentation.

Reply:
Also please add TC for multiple source send as well.